odl/phantom/geometric.py

```diff
@@ -611,14 +650,62 @@ def ellipsoid_phantom(space, ellipsoids):
     odl.phantom.geometric.defrise_ellipses : Ellipses for the
         Defrise phantom
     """
-
     if space.ndim == 2:
-        return _ellipse_phantom_2d(space, ellipsoids)
+        _phantom = _ellipse_phantom_2d
     elif space.ndim == 3:
-        return _ellipsoid_phantom_3d(space, ellipsoids)
+        _phantom = _ellipsoid_phantom_3d
     else:
         raise ValueError('dimension not 2 or 3, no phantom available')
 
+    if min_pt is None and max_pt is None:
+        return _phantom(space, ellipsoids)
+
+    else:
+        # Generate a temporary space with given `min_pt` and `max_pt`
```

Comment:
Is this truly the best way to implement this? Can't we change the function to do this properly?

Reply:
Yes, I tried internal first, but the ellipses are all cropped because the bounding box calculation is wrong. I don't have an overview about what exactly it does, so I don't know (and don't want to invest loads of time into finding out) what exactly happens. This implementation is way simpler.

Reply:
> Well, are we 100% sure we prefer this?

Not necessarily, but I also don't want to go into the low-level code that takes care of computing the bounding boxes for the ellipses and find out what part assumes that we look at the whole space all the time, and how to change it.

I honestly think that this solution is very acceptable and not a hack, just think about the logic of it: you create the fully sized phantom in the restricted (or shifted) space, and then you embed it into the original space. The only reason I'm not using a `ResizingOperator` or similar is that you don't want to make users hit the cell boundaries exactly with their `min_pt` and `max_pt`.